Added a new Hibernation test

juhlee-microsoft opened the pull request:
Hibernation patches are all accepted in linux-next. 
TestParameter hb_Branch and hb_url can accept linux-next or developer's repo info
Ubuntu 18.04 passed the test, but RHEL 7.7 failed due to disk space full in the middle of git-cloning. RHEL has different disk layout from the OS installation. None of small space is available for linux-next or developer's linux repo cloning. SUSE has package repo access issue, so no test result.

Target kernel is 5.5.0+ or 5.6.0.

[LISAv2 Test Results Summary]
Test Run On           : 02/13/2020 23:01:25
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Initial Kernel Version: 5.0.0-1031-azure
Final Kernel Version  : 5.5.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:43

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                Power-hibernate                                                                   PASS                40.02


Logs can be found at C:\LISAv2\TW37\TestResults\2020-13-02-15-01-20-4328

juhlee-microsoft committed:
hibernate test initial check-in

New PowerShell script along with a bash script.
New XML file with the updated ReplaceableTestParameters

juhlee-microsoft committed:
Fixed the error from debug

juhlee-microsoft committed:
moved the code to bash

juhlee-microsoft committed:
Merge branch 'master' into juhlee/hb1

juhlee-microsoft committed:
Fixed the error

juhlee-microsoft committed:
Clean up the logic

juhlee-microsoft committed:
Changed the logging

juhlee-microsoft committed:
Move req packages

juhlee-microsoft committed:
Fix for RHEL/CentOS

juhlee-microsoft committed:
fixed

juhlee-microsoft committed:
Update FunctionalTests-Power.xml

juhlee-microsoft committed:
Fixed

juhlee-microsoft committed:
Update SetupHbKernel.sh

juhlee-microsoft committed:
Update Power-Hibernate.ps1

juhlee-microsoft committed:
Added a new test script for both Powershell
script and bash script.
Created a new OneVMhb for Hibernation test.
Indentation fix from PR Review

juhlee-microsoft committed:
Update ReplaceableTestParameters.xml

juhlee-microsoft committed:
Update ReplaceableTestParameters.xml

juhlee-microsoft committed:
Merge branch 'juhlee/hb1' of https://github.com/LIS/LISAv2 into juhlee/hb1

dcui reviewed (commented):
When the VM suspends/resumes, maybe we should increase the wait time before checking the VM's status. Currently it's 2 min and 3 min respectively. Maybe we can check the status every 30 seconds and use a timeout of 30 minutes(???)

I had a quick look at the commit and it looks good to me (I'm not an expert on bash and power shell, though). Lili and Baihua should be able to provide more comments in detail.

juhlee-microsoft committed:
Changed for kernel compilation conditionally

Move this condition inside bash script

juhlee-microsoft committed:
Added timer in VM resume case

15 min timeout, and 15 second internval for state checking

juhlee-microsoft committed:
Update SetupHbKernel.sh

juhlee-microsoft committed:
Update Power-Hibernate.ps1

juhlee-microsoft committed:
Change based on the feedback

Try with the existing OneVM type. Will follow up for the cleanup later.

juhlee-microsoft committed:
Update Power-Hibernate.ps1

Fixed the typo
Change the format of the variable in Write-LogInfo function call.
Indentation

juhlee-microsoft committed:
Update SetupHbKernel.sh

Fixed based on PR review

juhlee-microsoft committed:
Merge branch 'juhlee/hb1' of https://github.com/LIS/LISAv2 into juhlee/hb1

juhlee-microsoft committed:
Removed the duplicated def

Leftover artifact

juhlee-microsoft committed:
Update FunctionalTests-Power.xml

juhlee-microsoft committed:
Added Hyper-v support

juhlee-microsoft committed:
Update Power-Hibernate.ps1

Changed per feedback
